content/200-concepts/100-components/drivers.mdx

```diff
@@ -0,0 +1,134 @@
+
+<!-- TODO: add image here -->
+
+## Database drivers
+
+Prisma Client can connect and run queries against your database using other JavaScript database drivers using **driver adapters**. Adapters act as _translators_ between Prisma Client and the JavaScript database driver. Prisma will use the Query Engine to transform the Prisma Client query to SQL and run the generated SQL queries via the JavaScript database driver.
+
+To use this feature:
+
+1. Update the `previewFeatures` block in your schema to include the the `driverAdapter` preview feature:
+
+   ```prisma
+   generator client {
+     provider        = "prisma-client-js"
+     previewFeatures = ["driverAdapters"]
+   }
+   ```
+
+2. Generate Prisma Client:
+
+   ```sh
+   npx prisma generate
+   ```
+
+## Serverless drivers
+
+By default, JavaScript database drivers use TCP connections. Several database providers enable use of different protocols, such as HTTP and WebSocket, to connect to your database. Example database providers include Neon, PlanetScale and Turso.
+
+Prisma maintains adapters for the following serverless database providers:
+
+- [Neon](#neon)
+- [PlanetScale](#planetscale)
+- [Turso](#turso)
+
+The following is a list of community maintained driver adapters:
+
+- [TiDB](https://github.com/tidbcloud/prisma-adapter)
+
+### Neon
+
+Install the Prisma adapter for Neon, Neon serverless driver and ws packages:
+
+```sh
+npm install @prisma/adapter-neon @neondatabase/serverless ws
+```
+
+Update your Prisma Client instance:
+
+```ts
+import { Pool, neonConfig } from '@neondatabase/serverless'
+import { PrismaNeon } from '@prisma/adapter-neon'
+import { PrismaClient } from '@prisma/client'
+import dotenv from 'dotenv'
+import ws from 'ws'
+
+dotenv.config()
+neonConfig.webSocketConstructor = ws
+const connectionString = `${process.env.DATABASE_URL}`
+
+const pool = new Pool({ connectionString })
+const adapter = new PrismaNeon(pool)
+const prisma = new PrismaClient({ adapter })
+```
+
+<!-- TODO: feedback link -->
+
+### PlanetScale
+
+Install the Prisma adapter for PlanetScale, PlanetScale serverless driver and undici packages:
+
+```sh
+npm install @prisma/adapter-planetscale @planetscale/database undici
+```
+
+Update your Prisma Client instance:
+
+```ts
+import { connect } from '@planetscale/database'
+import { PrismaPlanetScale } from '@prisma/adapter-planetscale'
+import { PrismaClient } from '@prisma/client'
+import dotenv from 'dotenv'
+import { fetch as undiciFetch } from 'undici'
+
+dotenv.config()
+const connectionString = `${process.env.DATABASE_URL}`
+
+const connection = connect({ url: connectionString, fetch: undiciFetch })
+const adapter = new PrismaPlanetScale(connection)
+const prisma = new PrismaClient({ adapter })
+```
+
+<!-- TODO: feedback link -->
+
+### Turso
```

Comment:
Turso is not the same as PlanetScale and Neon, as it is the _only_ way to talk to Turso.
Cloudflare D1 will be similar to Turso when we have it in the future, as will some others.